packages/-ember-data/addon/setup-container.js

```diff
@@ -46,9 +46,24 @@ function initializeDataAdapter(application) {
 }
 
 function initializeStoreInjections(application) {
-  application.inject('controller', 'store', 'service:store');
-  application.inject('route', 'store', 'service:store');
-  application.inject('data-adapter', 'store', 'service:store');
+  if (DEBUG && !application.inject) {
+    deprecate(
```

Comment:
I don’t think we need to also deprecate here as the other deprecation will always be thrown simultaneously. No sense throwing it twice.